lib/magma/actions/model_update_actions.rb

```diff
@@ -1,26 +1,71 @@
+require_relative 'base_action'
 require_relative 'update_attribute'
+require_relative 'add_attribute'
 
 class Magma
   class ModelUpdateActions
+    class FailedActionError < StandardError; end
+
     def self.build(project_name, actions_list)
       new(project_name, actions_list)
     end
 
     def perform
-      valid? && @actions.all?(&:perform)
+      return false unless valid?
+
+      Magma.instance.db.transaction do
+        raise FailedActionError unless @actions.all?(&:perform)
+        update_model_tables
+        true
+      end
+    rescue
```

Comment:
I let this rescue `StandardError` because I'm not sure what errors we'll get if migrations fail to run. I couldn't figure how to run an action that would cause a migration to fail.